File: code/py/modular_api/example/health/always_pass_health_check.py
from __future__ import annotations

from modular_api import HealthCheck, HealthCheckResult, HealthStatus


class AlwaysPassHealthCheck(HealthCheck):
    @property
    def name(self) -> str:
        return "example"

    async def check(self) -> HealthCheckResult:
        return HealthCheckResult(status=HealthStatus.PASS)

File: code/dart/modular_api/example/health/always_pass_health_check.dart
// ─── Example Health Check ─────────────────────────────────────────────────────
// In a real project you'd check a database connection, external service, etc.

import 'package:modular_api/modular_api.dart';

class AlwaysPassHealthCheck extends HealthCheck {
  @override
  final String name = 'example';

  @override
  Future<HealthCheckResult> check() async {
    return HealthCheckResult(status: HealthStatus.pass);
  }
}
